### PR TITLE
BeamSpot DQM clients modified to log to onlineDB only when beam status from TCDS is STABLE

### PR DESCRIPTION
**Pull request by dzuolo**
Following discussion with @ggovi BeamSpot DQM clients have been modified in such a way that they log to onlineDB only when the beam mode from TCDS is STABLE. This prevents the generation of dummy logs containing only "fit failed" reports during, e.g., cosmics data taking.

Code in this PR compile and the following local tests were successful 

-scram b code-format
-scram b code-checks
-scramv1 b runtests

This PR is not a backport and a backport to 12_3 will provided shortly.

FYI: @gennai @francescobrivio

### CODE REVIEW

**Comment by jfernan2** [comment]
test parameters:
- addpkg = DQM/Integration

**Comment by dzuolo** [comment]
@cmsbuild please test

**Comment by dzuolo** [comment]
Apparently I am not an "authorised" user that can issue the test command so if somebody could it for me I would appreciate it :)

**Comment by mmusich** [comment]
please test

File: DQM/BeamMonitor/plugins/BeamMonitor.cc [hunk 1]
```diff
@@ -661,6 +664,12 @@ void BeamMonitor::beginLuminosityBlock(const LuminosityBlock& lumiSeg, const Eve
 
 // ----------------------------------------------------------
 void BeamMonitor::analyze(const Event& iEvent, const EventSetup& iSetup) {
+  Handle<TCDSRecord> tcdsData;
+  iEvent.getByToken(tcdsToken_, tcdsData);
```
Comment (mmusich) [inline DQM/BeamMonitor/plugins/BeamMonitor.cc:668]:
```suggestion
  const TCDSRecord& tcdsData = iEvent.get(tcdsToken_);
```
as per https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#get

**Review by mmusich — COMMENTED** [review]
minor comment

**Comment by francescobrivio** [comment]
@dzuolo I think you need to modify also `beamhlt_dqm_sourceclient-live_cfg.py` which calls the same plugin (`BeamMonitor`).

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
> I think you need to modify also beamhlt_dqm_sourceclient-live_cfg.py which calls the same plugin (BeamMonitor).

apparently it's not tested (that's why it didn't fail..)

https://github.com/cms-sw/cmssw/blob/9bd0dbd0474b5546afa1b8e4d3240989663d7135/DQM/Integration/test/BuildFile.xml#L28-L29

given we somehow expect it to be the workhorse for Run3, can we find a way of giving it an HLT-stream like input?

**Comment by francescobrivio** [comment]
> given we somehow expect it to be the workhorse for Run3, can we find a way of giving it an HLT-stream like input? @francescobrivio @cms-sw/dqm-l2

Good point, I'll try to think of a way to re-enable the unitTest!

**Comment by dzuolo** [comment]
If I understood correctly the failed unit test is not related to this PR and is being discussed in https://github.com/cms-sw/cmssw/issues/37584. No other errors popped up.

**Comment by francescobrivio** [comment]
> > I think you need to modify also beamhlt_dqm_sourceclient-live_cfg.py which calls the same plugin (BeamMonitor).
> 
> apparently it's not tested (that's why it didn't fail..)
> 
> https://github.com/cms-sw/cmssw/blob/9bd0dbd0474b5546afa1b8e4d3240989663d7135/DQM/Integration/test/BuildFile.xml#L28-L29
> 
> given we somehow expect it to be the workhorse for Run3, can we find a way of giving it an HLT-stream like input? @francescobrivio @cms-sw/dqm-l2

Ok I think I have a working solution for this...I just need to get hold of a streamer file for `streamDQMOnlineBeamspot` stream. 😂 
Once I get that from DQM team I'll open a different PR to add the proper unit test.

**Comment by francescobrivio** [comment]
@emanueleusai given that you signed the 12_3_X backport can you sign this one as well?
The tests are clean except the known failure with `TestDQMOnlineClient-l1tstage2_dqm_sourceclient` (see #37584)

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
please test
(a temporary fix for the TestDQMOnlineClient-l1tstage2_dqm_sourceclient failure was made available in the IBs with #37634)

**Comment by perrotta** [comment]
+1

**Pull request merged**